spatialdata/_core/_spatialdata_ops.py

```diff
@@ -291,154 +287,76 @@ def _describe_paths(paths: list[list[Union[int, str]]]) -> str:
-                # this code should be never reached because the validate function should have raised an exception, but
-                # let's be extra safe
-                raise RuntimeError("Tables have incompatible region keys")
-            if isinstance(region, list):
-                all_region_keys.append(region_key)
-            else:
-                assert (
-                    len(all_region_keys) == 0
-                    or len(set(all_region_keys)) == 1
-                    and all_region_keys[0] == MERGED_TABLES_REGION_KEY
-                )
-                table.obs[MERGED_TABLES_REGION_KEY] = region
-                all_region_keys.append(MERGED_TABLES_REGION_KEY)
-        if len(set(all_region_keys)) != 1:
-            raise RuntimeError("Tables have incompatible region keys")
-        merged_region_key = all_region_keys[0]
-
-        all_regions = []
-        for table in tables:
-            region = table.uns[TableModel.ATTRS_KEY][TableModel.REGION_KEY]
-            if isinstance(region, str):
-                all_regions.append(region)
-            else:
-                all_regions.extend(region)
-        all_regions = list(set(all_regions))
-        merged_region = all_regions
-
-    attr = {"region": merged_region, "region_key": merged_region_key, "instance_key": merged_instance_key}
-    merged_table = anndata.concat(tables, join="outer", uns_merge="same")
-
-    # remove the MERGED_TABLES_REGION_KEY column if it has been added (the code above either adds that column
-    # to all the tables, either it doesn't add it at all)
-    for table in tables:
-        if MERGED_TABLES_REGION_KEY in table.obs:
-            del table.obs[MERGED_TABLES_REGION_KEY]
-
-    merged_table.uns[TableModel.ATTRS_KEY] = attr
-    merged_table.obs[merged_region_key] = merged_table.obs[merged_region_key].astype("category")
-    TableModel().validate(merged_table)
-    return merged_table
-
-
-def concatenate(sdatas: list[SpatialData], omit_table: bool = False) -> SpatialData:
-    """Concatenate a list of spatial data objects.
+            tables_l.append(table)
+            regions_l.append(regs)
+            instance_keys_l.append(table.obs[i].values.flatten())
+
+    merged_table = anndata.concat(tables_l, label=region_key, keys=regions_l, **kwargs)
+    merged_table.obs[instance_key] = np.concatenate(instance_keys_l, axis=0)
+    attrs = {
+        TableModel.REGION_KEY: regions_l,
+        TableModel.REGION_KEY_KEY: region_key,
+        TableModel.INSTANCE_KEY: instance_key,
+    }
+    merged_table.uns[TableModel.ATTRS_KEY] = attrs
+
+    return TableModel().validate(merged_table)
+
+
+def concatenate(
+    sdatas: list[SpatialData],
+    region_key: str = "region_merged",
+    instance_key: str = "instance_merged",
+    **kwargs: Any,
+) -> SpatialData:
+    """
+    Concatenate a list of spatial data objects.
 
     Parameters
     ----------
     sdatas
         The spatial data objects to concatenate.
-    omit_table
```

Comment:
I would keep this. Use case: we have a xenium and a visium dataset and we want to merge them keeping only the table of one of the two. To do so we can call `concatenate(..., omit_table=True)` and then do `sdata_concatenated.table = xenium.table`.

Reply:
what's the difference between manually adding the visium image and visium shapes to the xenium spatialdata?

Reply:
A big one: if you add an image to the xenium data, the new layer is saved to disk. On the contrary, `concatenate()` creates a new in-memory `sdata` object that is not backed (but the elements are backed from their respective `sdata` object)

Reply:
ok what about then simply passing one of the two `spatialdata` without the table? to me it just complicates the signature while the behavior is already impliclty implemetned

Reply:
> A big one: if you add an image to the xenium data, the new layer is saved to disk. On the contrary, concatenate() creates a new in-memory sdata object that is not backed (but the elements are backed from their respective sdata object)

this hidden behavior makes me also think we might want to be explicit about in memory/backed in concatenate

Reply:
I would remove this. I don't think the feature being described fits the definition of concatenation. Maybe this is more of a `"merge"`?

Also, is super easy for a user to do themselves in the meantime.

<details>
<summary> example </summary>

```python
from functools import reduce
from operator import or_

sdata_w_table = ...
other_sdatas: list[SpatialData]

sdatas = [sdata_w_table, *other_sdatas]

new_sdata = SpatialData(
    table=sdata_w_table.table,
    images=reduce(or_, (sdata.images for sdata in sdatas)),
    labels=reduce(or_, (sdata.labels for sdata in sdatas)),
    shapes=reduce(or_, (sdata.shapes for sdata in sdatas)),
    points=reduce(or_, (sdata.points for sdata in sdatas)),
)
```

</details>